### PR TITLE
LIMSA-72: Validation of the test plan & test unit in the edit mode 

### PR DESCRIPTION
**Pull request by weezyjr**
```
#30 New: orders Test plan /test unit validation in edit mode ... 
	
2020-01-22 10:19:28.291 | INFO     | ui_testing.testcases.base_test:setUp:38 - Test case : test017_validate_order_test_unit_test_plan_edit_mode
2020-01-22 10:19:52.891 | INFO     | api_testing.apis.base_api:info:66 - Get authorized api session.
2020-01-22 10:19:53.800 | INFO     | api_testing.apis.base_api:info:66 - session ID : eyJhbGciOi .....
2020-01-22 10:19:53.806 | INFO     | ui_testing.pages.login_page:login:10 - Login admin : admin.
2020-01-22 10:20:30.305 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-01-22 10:20:35.307 | INFO     | ui_testing.testcases.basic_tests.test06_orders:test017_validate_order_test_unit_test_plan_edit_mode:665 -  Running test case to check that at least test unit or test plan is mandatory in order
2020-01-22 10:20:38.490 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-01-22 10:20:43.498 | INFO     | ui_testing.pages.orders_page:get_random_order:68 -  + Get random order.
2020-01-22 10:20:45.258 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-01-22 10:20:50.273 | INFO     | ui_testing.testcases.basic_tests.test06_orders:test017_validate_order_test_unit_test_plan_edit_mode:671 -  + order_url : https://automation.1lims.com/sample/orders/489
2020-01-22 10:20:50.274 | INFO     | ui_testing.testcases.basic_tests.test06_orders:test017_validate_order_test_unit_test_plan_edit_mode:673 -  Remove all selected test plans and test units
2020-01-22 10:20:50.864 | INFO     | ui_testing.pages.base_pages:confirm_popup:71 - Confirming the popup
2020-01-22 10:21:01.145 | INFO     | ui_testing.pages.base_pages:save:55 - save the changes
2020-01-22 10:21:11.505 | INFO     | ui_testing.testcases.base_test:tearDown:67 - TearDown. 	
ok

----------------------------------------------------------------------
XML: /home/modeso15/Documents/1lims/1lims-automation/nosetests.xml
----------------------------------------------------------------------
Ran 1 test in 103.219s

OK
```

### CODE REVIEW

File: config.ini [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [site]
-#url=https://testing.1lims.com/
-url=https://automation.1lims.com/
+url=https://testing.1lims.com/
+#url=https://automation.1lims.com/
```
Comment (ghada-37) [inline config.ini]: we can run it on automation instance now

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 2]
```diff
@@ -665,39 +665,50 @@ def test016_validate_order_test_unit_test_plan(self):
...
-        self.base_selenium.LOGGER.info(' + order_url : {}'.format(order_url))
+        self.base_selenium.LOGGER.info(' Running test case to check that at least test unit or test plan is mandatory in order')
+
+        # Get random order
+        order_request = self.orders_api.get_all_orders(limit=20).json()
+        self.assertEqual(order_request['status'], 1)
+        orders_records = order_request['orders']
+        self.assertNotEqual(len(orders_records), 0)
+        random_order_index = self.generate_random_number(lower=0, upper=len(orders_records) - 1)
+        selected_order_record = orders_records[random_order_index]
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py]: you can use 
`selected_order_record = random .choice(order_request['orders']) 
`directly

File: config.ini (hunk 1 above)
Comment (dianaeldemerdash) [reply config.ini:3]: Done

**Comment by dianaeldemerdash** [comment]
nosetests3 -vs --logging-level=WARNING ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test017_validate_order_test_unit_test_plan_edit_mode --tc-file=config.ini --tc=site.password:admin 
2020-04-30 05:16:42.583 | INFO     | api_testing.apis.base_api:info:76 - Get authorized api session.
2020-04-30 05:16:43.314 | INFO     | api_testing.apis.base_api:info:76 - session ID : eyJhbGciOi .....
New: orders Test plan /test unit validation in edit mode ...    
2020-04-30 05:16:43.508 | INFO     | ui_testing.testcases.base_test:info:129 - Test case : test017_validate_order_test_unit_test_plan_edit_mode
2020-04-30 05:17:14.010 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-04-30 05:17:15.323 | INFO     | ui_testing.testcases.basic_tests.test06_orders:test017_validate_order_test_unit_test_plan_edit_mode:611 -  Running test case to check that at least test unit or test plan is mandatory in order
2020-04-30 05:17:15.323 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders
2020-04-30 05:17:16.054 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-04-30 05:17:17.727 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-04-30 05:17:22.734 | INFO     | ui_testing.testcases.basic_tests.test06_orders:test017_validate_order_test_unit_test_plan_edit_mode:618 -  Remove all selected test plans and test units
2020-04-30 05:17:23.006 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-04-30 05:17:28.540 | INFO     | ui_testing.pages.base_pages:confirm_popup:73 - confirming the popup
2020-04-30 05:17:28.753 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-04-30 05:17:33.893 | INFO     | ui_testing.pages.base_pages:info:274 -  save the changes
2020-04-30 05:17:33.941 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-04-30 05:17:36.165 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-04-30 05:17:38.371 | INFO     | ui_testing.testcases.base_test:info:129 - TearDown.        
ok

----------------------------------------------------------------------
Ran 1 test in 54.864s

OK

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 2 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py]: Done

**Pull request merged**